osi3trace/osi_trace.py

```diff
@@ -57,7 +210,7 @@ def __init__(self, path=None, type_name="SensorView", cache_messages=False):
 
     def from_file(self, path, type_name="SensorView", cache_messages=False):
         """Import a trace from a file"""
-        self.type = self.map_message_type(type_name)
+        self.type = OSITrace.map_message_type(type_name)
 
         if path.lower().endswith((".lzma", ".xz")):
```

Comment:
Keep to same logic as original dispatch, also since path is not guaranteed to be a Path:

```suggestion
        if path.suffix.lower() in [".lzma", ".xz"]:
```